src/OrchardCore/OrchardCore.DisplayManagement.Liquid/LiquidViewTemplate.cs

```diff
@@ -56,7 +56,7 @@ internal static async Task RenderAsync(RazorPage<dynamic> page)
                 var content = new ViewBufferTextWriterContent();
                 await context.EnterScopeAsync(page.ViewContext, (object)page.Model);
                 await template.FluidTemplate.RenderAsync(content, htmlEncoder, context);
-                content.WriteTo(page.Output, htmlEncoder);
+                page.Output.Write(content);
```

Comment:
When calling `content.WriteTo(page.Output)` our implementation goes over each chunk of the internal string builders and calls Write(ReadOnlySpan), which is not implemented in aspnet core https://github.com/dotnet/aspnetcore/blob/main/src/Mvc/Mvc.ViewFeatures/src/Buffers/ViewBufferTextWriter.cs

So the TextWriter implementation is used, and it rents an array and then calls Write(char[]), which in `ViewBufferTextWriter` does a string conversion.

By doing `page.Otuput.Write(content)` it's now invoking `ViewBufferTextWriter.Write(IHtmlContent)` which doesn't allocate anything. And then when the `IHtmlContent` is used, it should write directly to the response buffer (haven't verified this theory).

Reply:
> which is not implemented in aspnet core

The base implementation of `TextWriter` does this

https://github.com/dotnet/runtime/blob/2810ae5ff692108291692dec5ac36ba698af887b/src/libraries/System.Private.CoreLib/src/System/IO/TextWriter.cs#L183-L196

        public virtual void Write(ReadOnlySpan<char> buffer)
        {
            char[] array = ArrayPool<char>.Shared.Rent(buffer.Length);

            try
            {
                buffer.CopyTo(new Span<char>(array));
                Write(array, 0, buffer.Length);
            }
            finally
            {
                ArrayPool<char>.Shared.Return(array);
            }
        }

Reply:
That's my second phrase, exactly.

Reply:
Okay cool then